posthog/api/survey.py

```diff
@@ -751,6 +751,234 @@ def responses_count(self, request: request.Request, **kwargs):
 
         return Response(counts)
 
+    def _validate_dates(self, date_from: str | None, date_to: str | None) -> None:
+        """Validate that date_from and date_to are in ISO 8601 format.
+
+        Args:
+            date_from: Optional ISO timestamp for start date
+            date_to: Optional ISO timestamp for end date
+
+        Raises:
+            ValidationError: If dates are not in ISO 8601 format
+        """
+        if date_from or date_to:
+            try:
+                if date_from:
+                    datetime.fromisoformat(date_from.replace("Z", "+00:00"))
+                if date_to:
+                    datetime.fromisoformat(date_to.replace("Z", "+00:00"))
+            except ValueError:
+                raise exceptions.ValidationError(
+                    "Invalid date format. Please use ISO 8601 format (e.g. 2024-01-01T00:00:00Z)"
+                )
```

Comment:
this is doing extra checks at runtime, up to 4x instead of 2x
maybe extract to a method or just 2 try catch?